src/services/heartbeatListener/heartbeatListenerService.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Heartbeat Listener Service
+ * This is an example service that listens for heartbeat events that indicate that windows
+ * are in a bad state. It performs the following actions:
+ * - Removes a window from a stack if that window enters the state "notResponding"
+ * - Kills a window if it enters the state "crashed"
+ * 
+ * Available heartbeat events (times to reach state are configurable):
+ * "notResponding"
+ * "possiblyCrashed"
+ * "crashed"
+ * "responding" (only when returning to a good state after being in one of the above states)
+ */
+
+const Finsemble = require("@chartiq/finsemble");
+const RouterClient = Finsemble.Clients.RouterClient;
+const BaseService = Finsemble.baseService;
+const Logger = Finsemble.Clients.Logger;
+
+class heartbeatListenerService extends BaseService {
+
+	constructor() {
+		super();
+		
+		this.handleHeartbeat = this.handleHeartbeat.bind(this);
+	}
+	/**
+	 * Responds to a HeartbeatTimeout Event
+	 * - Kicks windows out of a stack if they are not responding
+	 * - Kills crashed windows
+	 * @param {*} err 
+	 * @param {*} response 
+	 */
+	handleHeartbeat(err, response) {
+		//check if the window is in a stack, if so remove it, otherwise do nothing
+		if (response.data.type === "notResponding") {			

```

Comment:
Instead of an if series could we instead do a `switch` on the response type? This would allow us to have the other modes already provided with the default to return doing nothing. Making it clear the other modes are possible and where to modify them if needed in the future.